src/glang/codegen/builtin_types.py

```diff
@@ -122,30 +129,23 @@ def ir_mangle(self) -> str:
             assert self.alias is not None
             return self.alias.ir_mangle
 
-        prefix = "__T_"
-        if len(self.specialization) == 0:
-            return prefix + self.name
+        # Perhaps built-in types shouldn't be a NamedType.
+        if isinstance(self.definition, PrimitiveDefinition):
+            return self.definition.ir_mangle
```

Comment:
On the other hand, we could output our names (e.g. `i32`), but then it becomes harder to call Graphene code from C/C++ land.

Reply:
That sounds reasonable, although it's a bit awkward to implement as the types would have to be defined as structs - if it's a `typedef` or `using` then it ends up mangling it as the built-in type, instead of using the name you gave it: https://godbolt.org/z/1x3qaKj1o.

Anyway, we can change this later if we ever get around to writing C++ bindings :)